app/assets/javascripts/controllers/service/service_form_controller.js

```diff
@@ -0,0 +1,66 @@
+ManageIQ.angular.app.controller('serviceFormController', ['$http', '$scope', 'serviceFormId', 'miqService', function($http, $scope, serviceFormId, miqService) {
+    var init = function() {
+
+      $scope.serviceModel = {
+        name: '',
+        description: ''
+      };
+      $scope.formId = serviceFormId;
+      $scope.afterGet = false;
+      $scope.newRecord = false;
+      $scope.modelCopy = angular.copy( $scope.serviceModel );
+      ManageIQ.angular.scope = $scope;
+
+      miqService.sparkleOn();
+      $http.get('/service/service_form_fields/' + serviceFormId).success(function(data) {
+        $scope.serviceModel.name        = data.name;
+        $scope.serviceModel.description = data.description;
+
+        $scope.afterGet = true;
+        $scope.modelCopy = angular.copy( $scope.serviceModel );
+        miqService.sparkleOff();
+      });
+
+      $scope.$watch("serviceModel.name", function() {
+        $scope.form = $scope.angularForm;
+        $scope.model = "serviceModel";
+      });
+    };
+
+    var serviceEditButtonClicked = function(buttonName, serializeFields) {
+      miqService.sparkleOn();
+      var url = '/service/service_edit/' + serviceFormId + '?button=' + buttonName;
+      
+      if (serializeFields === undefined) {
```

Comment:
This condition could be replaced by `miqService.miqAjaxButton(url, serializeFields);`, if it's undefined, there's no need to *not* pass it, and the end result is the same.